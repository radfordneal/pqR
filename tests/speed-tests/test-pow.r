source("time.r")
 
test.name <- "pow.scalar"

b <- 0.7

f <- test.cmp(function(){ for (i in 1:3000000) a <- 2^b; a })
sys.time(a<-f())
print(a)

f <- test.cmp(function(){ for (i in 1:3000000) a <- b^2; a })
sys.time(a<-f())
print(a)


test.name <- "pow.vec-to-scalar"

x <- seq(1,2,length=100)
y <- seq(1,2,length=1000)
z <- seq(1,2,length=10000)

d2 <- 2
d1 <- 1
d0 <- 0
dm1 <- -1
d0.5 <- 0.5

f <- test.cmp(function(){ for (i in 1:3000) a <- z^0.3; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:3000) a <- z^d0.5; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:10000) a <- z^d1; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:10000) a <- z^d0; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:10000) a <- z^dm1; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:2000000) a <- x^2; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:1000000) a <- y^2; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:100000) a <- z^d2; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:100000) a <- z*z; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))


test.name <- "pow.vec-to-vec"

m <- seq(1,2,length=100)
o <- rep(c(1,2),length=100)
two <- rep(2.0,100)
oi <- rep(1:7,length=100)
xi <- rep(1:11,length=100)

f <- test.cmp(function(){ for (i in 1:100000) a <- xi^oi; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:100000) a <- x^m; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:1000000) a <- x^o; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))

f <- test.cmp(function(){ for (i in 1:2000000) a <- x^two; a})
sys.time(a<-f())
print(c(sum(a),sum(a^2)))
