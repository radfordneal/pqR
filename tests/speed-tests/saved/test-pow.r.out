
---- pow.scalar 

Time:
[1] 1.624505
Time:
[1] 0.49

---- pow.vec-to-scalar 

Time:
[1] 11248.37 12696.45
Time:
[1] 12189.5 15000.0
Time:
[1] 15000.0 23333.5
Time:
[1] 10000 10000
Time:
[1] 6931.529 5000.125
Time:
[1] 233.5017 622.3236
Time:
[1] 2333.500 6202.302
Time:
[1] 23333.5 62002.3
Time:
[1] 23333.5 62002.3

---- pow.vec-to-vec 

Time:
[1] 5.320280e+07 6.132989e+14
Time:
[1] 205.4990 494.6501
Time:
[1] 192.2559 430.9505
Time:
[1] 233.5017 622.3236
